### PR TITLE
Tic-Tac-Toe Milestone 2: User Interface!

### PR DESCRIPTION
**Pull request by ferbaco86**
On this milestone, we are creating the basic user interface of the game.
## Project milestones
- Create a specific branch with name: user_interface.
- main.rb file should be responsible for user interaction. In this milestone you should only create content in this file.
- Display prompts for users to create a basic game flow
- Make sure that you can run your game with your executable file and play with the basic flow.
- The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe.

Captura de Pantalla 2020-05-01 a la(s) 11 28 18

## General requirements:
- [x] Add a descriptive README file to your repositories as described here.
- [x] Use meaningful names for your Pull Requests and add a short summary of its content as described here.
- [x] Use descriptive commit messages as described here.
- [x] Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- [x] Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.
## Git Flow:
- [x] Use a specific branch for each feature.
- [x] Use a descriptive name for your branches.
- [x] Use Rubocop to validate your code.

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
# Hi team! :clap: 
### Good job setting up the project you are missing just one requirement  in Milestone 1 :eyes: 

*  In your project should be `bin` directory with `main.rb` file  from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)
screenshot
*  `main.rb` is [executable file](https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/):

* To Test if this executable file works with next command: 
`bin/main.rb` or `ruby bin/main.rb` and it should displays anything.


##### Behaviour

* [ ] In this step not everything must work, the point is to make sure that students thought about everything listed below. They might do it only with inline comments from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/). 
screenshot 
* [ ] Basic game flow is prepared : 

  
  * [ ] `[REQUIRED]` game informs player if selected move is invalid
  * [ ] `[REQUIRED]` game informs player if selected move is a winning move
  * [ ] `[REQUIRED]` game informs player if selected move is a draw move
  * [ ] `[REQUIRED]` game repeats all actions for next player's move


### [SUGGESTION] The structure of the game's loop should be as follows:

##### Implementation details

* [ ] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [ ] Inside loop, you can all actions repeated for each move
* [ ] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

Using out-of-nowhere numbers to terminate a loop, for example 5.times do ... end is a signal that something is wrong. We recommend the usage of a terminating variable (while game_on) 

### *The final point of this milestone is to make sure that you thought about all the actions needed to play Tic Tac Toe from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)


### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Comment by ferbaco86** [comment]
## [SUGGESTION] The structure of the game's loop should be as follows:
Implementation details

- [x] A loop is needed in this game (recommended is while but anything else will do) (OdinProject hint)

- [x] Inside loop, you can all actions repeated for each move

- [x] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

Using out-of-nowhere numbers to terminate a loop, for example 5.times do ... end is a signal that something is wrong. We recommend the usage of a terminating variable (while game_on)


> Hi, thanks a lot for the guidelines on how to structure the game flow on this branch. The instructions on the Pathwright do not give clear instructions on how to achieve this.

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+puts "#{name_player1} will have the X"
+puts "#{name_player2} will have the O"
+puts ''
+puts 'HERE THE BOARD WILL BE DRAWN'
+puts ''
+puts "#{name_player1} will have the first move"
+puts ''
+round_num = 0
+
+while round_num < 9
```
Comment (shshamim63) [inline bin/main:22]: [Suggestion]
It is better not to use a loop based on a sudden number. Cause it reduces the scalability of a task. So I will suggest you not to use it in your game_logic milestone.

**Review by shshamim63 — APPROVED** [review]
## Status :100: Approved :1st_place_medal: 
___
Hi @ferbaco86 and @lmaldonadoch :wave: ,
awesome job :clap: you have covered all the corner cases needed to organize the game. You have also added a Readme template, Nothing is blocking me now to approve this milestone. This :+1: is from me. you can go ahead.
___
:rocket: Why pathwright contains fewer instructions?
:point_right: Well, to ensure that you can play with any task under minimal supervision. Let me know if you think otherwise. 
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Pull request merged**